tests/darcy_velocity.prm

```diff
@@ -0,0 +1,147 @@
+  set Model name = function
+  subsection Function
+    set Function expression       = 293
+  end
+end
+
+subsection Boundary temperature model
+  set List of model names = box
+  set Fixed temperature boundary indicators   = top, bottom, left, right
+  subsection Box
+    set Bottom temperature = 293
+    set Top temperature    = 293
+    set Left temperature   = 293
+    set Right temperature  = 293
+  end
+end
+
+subsection Boundary velocity model
+  set Prescribed velocity boundary indicators = left x:function, right x:function
+  set Tangential velocity boundary indicators = top, bottom
+  subsection Function
+    set Coordinate system = cartesian
+    set Variable names = x,y,t
+    set Function expression = 1;0
+  end
+end
+
+############### Compositional fields
+# We want to use three compositional fields, the porosity and
+# and two additional fields. The solid field will be advected with
+# the solid speed prescribed above, and the fluid field will be advected
+# as the sum of the solid velocity and the vertical buoyancy the fluid has.
+# The porosity field is uniform to allow for a uniform fluid velocity. The 
+# fluid field is a circle of radius 12.5km at the center of the model domain.
+subsection Compositional fields
+  set Number of fields = 3
+  set Names of fields = porosity, solid, fluid
+  set Compositional field methods = darcy field, field, darcy field
+  set Types of fields = porosity, generic, generic
+end
+
+subsection Initial composition model
+  set Model name = function
+  subsection Function
+    set Variable names      = x,y
+    set Function constants  = pi=3.1415926,x0=100000,y0=50000,c=12500, A_f=0.2, A_s = 1.0
+    set Function expression = A_f; if((x - x0)*(x - x0) + (y - y0)*(y - y0) <= c*c, A_s, 0); if((x - x0)*(x - x0) + (y - y0)*(y - y0) <= c*c, A_s, 0)
+  end
+end
+
+subsection Boundary composition model
+  set List of model names = initial composition
+  set Fixed composition boundary indicators = bottom, top, left, right
+end
+
+# We then choose a vertical gravity model and describe the
+# initial temperature with a vertical gradient.
+subsection Gravity model
+  set Model name = vertical
+  subsection Vertical
+    set Magnitude = 10.0
+  end
+end
+
+# The 'melting rate' material model from rising_melt_blob.cc is being
+# used since it provides terms for the fluid phase in this test. Melt
+# is not actually turned on, but the structure of the material model is 
+# being used to fill fluid related parameters.
+subsection Material model
+  set Model name = melting rate
+end
+
+
+# The final part of this input file describes how many times the
+# mesh is refined and what to do with the solution once computed
+subsection Mesh refinement
+  set Initial adaptive refinement        = 0
+  set Initial global refinement          = 5
+  set Time steps between mesh refinement = 0
+end
+
+subsection Postprocess
+  set List of postprocessors = composition statistics, visualization
```

Comment:
Looking at the output, it occurs to me that there is not actually anything that would allow us to detect if the test produces the correct results (we do not know u_f, and we only have the global integral of the compositional field, so we do not know where the blob is). I would add the "maximum depth of field" postprocessor here, which shows the largest depth at which a point is found where the field has a value of 0.5 or larger. This value should decrease over time for the fluid field.